presto-docs/src/main/sphinx/optimizer/statistics.rst

```diff
@@ -0,0 +1,144 @@
+
+
+The set of statistics available for a particular query depends on the connector being used and can also vary by table or
+even table layout. For example, the Hive connector does not currently provide statistics on data size.
+
+Displaying Table Statistics
+---------------------------
+
+Table statistics can be displayed via the Presto CLI using the ``SHOW STATS`` command.
+There are two flavors of the command:
+
+ * ``SHOW STATS FOR <table_name>`` will show statistics for the table layout representing all data in the table
+ * ``SHOW STATS FOR (SELECT <column_list|*> FROM <table_name> WHERE <filtering_condition>)``
+   will show statistics for the table layout of table ``t`` representing a subset of data after applying the given filtering
+   condition. Both the column list and the filtering condition used in the ``WHERE`` clause can reference table columns.
+
+In both cases, the ``SHOW STATS`` command outputs two types of rows.
+For each column in the table there is a row with ``column_name`` equal to the name of that column.
+These rows expose column-related statistics for a table (data size, nulls count, distinct values count).
+Additionally there is one row with NULL as the ``column_name``. This row contains table-layout wide statistics - for now just the row count.
+
+For example:
+
+.. code-block:: none
+
+    presto:default> SHOW STATS FOR nation;
+
+     column_name | data_size | distinct_values_count | nulls_count | row_count
+    -------------+-----------+-----------------------+-------------+-----------
+     comment     | NULL      |                  31.0 |        31.0 | NULL
+     nationkey   | NULL      |                  19.0 |        19.0 | NULL
+     name        | NULL      |                  24.0 |        24.0 | NULL
+     regionkey   | NULL      |                   5.0 |         5.0 | NULL
+     NULL        | NULL      | NULL                  | NULL        |      25.0
+     (5 rows)
+
+
+    presto:default> SHOW STATS FOR (SELECT * FROM nation WHERE nationkey > 10);
+
+     column_name | data_size | distinct_values_count | nulls_count | row_count
+    -------------+-----------+-----------------------+-------------+-----------
+     comment     | NULL      |                  21.0 |        21.0 | NULL
+     nationkey   | NULL      |                   9.0 |         9.0 | NULL
+     name        | NULL      |                  14.0 |        14.0 | NULL
+     regionkey   | NULL      |                   3.0 |         3.0 | NULL
+     NULL        | NULL      | NULL                  | NULL        |      15.0
+     (5 rows)
+
+If provided ``SELECT`` will filter out all of the partitions (all table layouts),
+then the ``SHOW STATS`` will return no statistic which will be represented as in example below.
+
+.. code-block:: none
+
+    presto:default> SHOW STATS FOR (SELECT * FROM nation WHERE nationkey > 999);
+
+     column_name
+    -------------
+     NULL
+    (1 row)
```

Comment:
* there is no `row_count` column in docs 
* row count should be 0 (in testing it showed up as null)
* IMO other columns should be reported too